fix: Misc style fixes (for Liquid)

vinnyjth opened the pull request:
## 🐛 Issue

https://3.basecamp.com/3926363/buckets/32694519/card_tables/cards/7009918938

https://3.basecamp.com/3926363/buckets/32694519/card_tables/cards/7010425769


## Solution

- Fixed search bar width, so it's no longer wider than content.
- Fixed the ActionList feature so it no longer truncates text too early
- Fixed margin around modal header

## Testing

For testing the search bar width, you can click on any item on desktop. Here's an example:

https://apollos-web-embeds-git-cleanup-header-apollos-embeds.vercel.app/?id=in-training-for-reigning-TWVkaWFDb250ZW50SXRlbS1kODI0Yjg0ZS00MDFkLTQ0NzAtOGFlNi1mYjU3Y2U4NWUwMmI%3D

For testing the action list, click on the search bar in mobile view :) 



## ✏️ Solution

<img width="1512" alt="CleanShot 2024-01-30 at 20 51 12@2x" src="https://github.com/ApollosProject/apollos-embeds/assets/1637694/8acb719f-a836-4861-a053-b5d0faed70b3">
<img width="259" alt="CleanShot 2024-01-30 at 20 51 00@2x" src="https://github.com/ApollosProject/apollos-embeds/assets/1637694/b1961384-1e50-4d87-b21b-d091f6a280b0">
<img width="878" alt="CleanShot 2024-01-30 at 20 39 31@2x" src="https://github.com/ApollosProject/apollos-embeds/assets/1637694/94ca2dd9-58b4-4cf3-b218-3b37f315f7f7">
<img width="855" alt="CleanShot 2024-01-30 at 20 38 52@2x" src="https://github.com/ApollosProject/apollos-embeds/assets/1637694/c4d8766c-6c6f-4a40-aaf6-29a4f11700e6">
<img width="892" alt="CleanShot 2024-01-30 at 20 38 42@2x" src="https://github.com/ApollosProject/apollos-embeds/assets/1637694/67e325e2-d37f-4cba-b9db-a5571ead9f77">
<img width="477" alt="CleanShot 2024-01-31 at 08 08 27@2x" src="https://github.com/ApollosProject/apollos-embeds/assets/1637694/f55f1ee1-6f43-406c-9c58-1c4f4279eaba">

nlewis84 commented on packages/web-shared/ui-kit/ResourceCard/ResourceCard.js:
```diff
@@ -89,10 +72,7 @@ function ResourceCard({
-                sm: '250px',
-                md: '350px',
-                lg: '500px',
+                _: '90%',
```
Maybe it's only a problem if there is only one `responsive` style and it should be `"90%"` in that case? `maxWidth` change up above may need that tweak also.

nlewis84 replied:
This caught my eye because I was trying to figure out why the action list item in your description picture was forcing the caret to the right when it had a long title.

vinnyjth replied:
@nlewis84 Were you able to recreate that carret being forced to the left? I think I fixed that but accidentally  in dragged in an old screenshot but I'll check again :) 

that's a good callout regarding the `[object Object]`. I think the code is working, in that the styles are being applied, but it's broken that the `prop` makes it all the way down to the dom. It should def be redirected into the stylesheet.

nlewis84 replied:
I could not recreate that issue 😄 
Not a blocking issue. This does seem to be working.

vinnyjth replied:
Cool! I just confirmed right now also. You're right that I don't need the responsive object and I'm cleaning that up now before we merge

nlewis84 reviewed (commented):
Just one thing that caught my eye. Otherwise, looks good!

nlewis84 reviewed (approved):
👍

vinnyjth commented:
cleaned up. and spotted a bug with the positioning of the arrow (it was too high) that I fixed

<img width="539" alt="CleanShot 2024-01-31 at 10 22 55@2x" src="https://github.com/ApollosProject/apollos-embeds/assets/1637694/64ccfb1d-e8da-4838-8c93-cf26657ac5f4">
